pkg/injector/patcher/sidecar_container.go

```diff
@@ -70,6 +70,9 @@ func (c *SidecarConfig) getSidecarContainer(opts getSidecarContainerOpts) (*core
 		"--app-protocol", c.AppProtocol,
 		"--log-level", c.LogLevel,
 		"--dapr-graceful-shutdown-seconds", strconv.Itoa(c.GracefulShutdownSeconds),
+		"--control-plane-namespace", c.sec.ControlPlaneNamespace(),
```

Comment:
This will make the injector unable to inject Dapr sidecars running older versions of Dapr, since they'll see an unsupported CLI flag and crash. The E2E tests will fail with this.

Reply:
How have flags been added in the past?

Reply:
This will help for the future, although can't help for today (we need to wait at least 2 more Dapr releases before we can leverage it): https://github.com/dapr/dapr/pull/6766

Reply:
I'm not a fan of ignoring unknown flags. Exiting error on unknown flags is standard for all programs, and makes things super confusing for users to know whether a CLI flag was even picked up. I don't want to think about the human hours wasted by not noticing a spelling mistake in a flag :grimacing: 


We can use environment variables for backwards compat until all supported versions have the CLI flag.